src/testRunner/unittests/customTransforms.ts

```diff
@@ -95,6 +95,39 @@ namespace ts {
             module: ModuleKind.ES2015,
             emitDecoratorMetadata: true,
             experimentalDecorators: true
-         });
+        });
+
+        emitsCorrectly("sourceMapExternalSourceFiles",
+            [
+                {
+                    file: "source.ts",
+                    // The text of length 'changed' is made to be on two lines so we know the line map change
+                    text: `\`multi
+                    line\`
+'change'`
+                },
+            ],
+            {
+                before: [
+                    context => node => visitNode(node, function visitor(node: Node): Node {
+                        if (isStringLiteral(node) && node.text === "change") {
+                            const text = "'changed'";
+                            const lineMap = computeLineStarts(text);
+                            setSourceMapRange(node, {
+                                pos: 0, end: text.length, source: {
+                                    text,
+                                    fileName: "another.html",
+                                    lineMap,
+                                    getLineAndCharacterOfPosition: pos => computeLineAndCharacterOfPosition(lineMap, pos)
+                                }
+                            });
+                            return node;
+                        }
+                        return visitEachChild(node, visitor, context);

```

Comment:
I appreciate that this doesn't actually affect the test of the source map source, but, to be complete, shouldn't the text value of the returned `node` be changed from `change` to `'changed'`?

Reply:
I am not changing the text, Just setting the sourcePosition to be different range (for test perspective its not needed to change the text, we have other tests above this that do that.)

Reply:
Yes, I agree that it does not affect the effectiveness of this test. It is just that it could be confusing to someone coming to this test in the future to find that the source mapping says that the third line in the generated file came from the `another.html` while the generated file still has the original text from the `source.ts` file. You can see this in its [visualization](http://sokra.github.io/source-map-visualization/#base64,Im11bHRpXG4gICAgICAgICAgICAgICAgICAgIGxpbmUiOwonY2hhbmdlJzs=,eyJ2ZXJzaW9uIjozLCJmaWxlIjoic291cmNlLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsic291cmNlLnRzIiwiYW5vdGhlci5odG1sIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLGlDQUN5QixDQUFBO0FDRHpCLFFBQVMsQ0RFRCJ9,YG11bHRpCiAgICAgICAgICAgICAgICAgICAgbGluZWAKJ2NoYW5nZSc=,J2NoYW5nZWQn)

Not a blocker but something that might make confuse someone in the long future.